zebra-chain/src/parameters/network_upgrade.rs

```diff
@@ -76,7 +76,7 @@ pub(crate) const TESTNET_ACTIVATION_HEIGHTS: &[(block::Height, NetworkUpgrade)]
 
 /// The Consensus Branch Id, used to bind transactions and blocks to a
 /// particular network upgrade.
-#[derive(Copy, Clone, Debug, Eq, Hash, PartialEq)]
+#[derive(Copy, Clone, Debug, Eq, Hash, PartialEq, Serialize, Deserialize)]
```

Comment:
If we switched to NetworkUpgrade, we'd replace the changes in this file with a new method on NetworkUpgrade, which:
- looks up the network upgrade for a u32 branch id, and
- returns a serialization error if the lookup fails